Python/sysmodule.c

```diff
@@ -374,6 +374,27 @@ sys_exc_info_impl(PyObject *module)
 }
 
 
+/*[clinic input]
+sys.unraisablehook
+
+    exc_type: object
+    exc_value: object
+    exc_tb: object
+    obj: object
+    /
+
+Print an unraisable exception.
+[clinic start generated code]*/
+
+static PyObject *
+sys_unraisablehook_impl(PyObject *module, PyObject *exc_type,
+                        PyObject *exc_value, PyObject *exc_tb, PyObject *obj)
+/*[clinic end generated code: output=f83f725d576ea3d3 input=2aba571133d31a0f]*/
+{
+    return _PyErr_WriteUnraisable(exc_type, exc_value, exc_tb, obj);
```

Comment:
Why not set `sys.unraisablehook` to None by default? Or not set it at all? What is the use case for exposing it at Python level?

Reply:
There is a difference between `sys.breakpointhook` and `sys.unraisablehook`. The former is only called when you explicitly call `breakpoint()`. The latter can be called implicitly at arbitrary place and time. It is out of your control. `sys.excepthook` is also called implicitly, but at known circumstances -- when an uncaught exception achieves the top level. I afraid that calling `sys.unraisablehook` at arbitrary place may be unsafe.

Reply:
sys.unraisablehook takes (exc_type, exc_value, exc_tb, obj) arguments, it doesn't rely on the current exception. Why would it be called to call it?

Reply:
Sorry: Why would it be **unsafe** to call it?

Reply:
For example because it can be called deep in the C stack.